### PR TITLE
various small documentation improvements

### PR DESCRIPTION
**Pull request by narimiran**

### CODE REVIEW

File: lib/pure/asyncdispatch.nim [hunk 1]
```diff
@@ -42,12 +42,12 @@
...
-##      future.addCallback(
-##        proc () =
-##          echo(future.read)
-##      )
+## .. code-block:: Nim
+##    var future = socket.recv(100)
+##    future.addCallback(
+##      proc () =
+##        echo(future.read)
+##    )
```
Comment (konsumlamm) [inline lib/pure/asyncdispatch.nim:50]: Why not just change it to `runnableExamples`? Ditto for the other code-blocks.

File: lib/pure/collections/tables.nim [hunk 2]
```diff
@@ -499,7 +501,8 @@ template tabCellHash(i)  = t.data[i].hcode
 proc del*[A, B](t: var Table[A, B], key: A) =
   ## Deletes `key` from hash table `t`. Does nothing if the key does not exist.
   ##
-  ## .. warning:: If duplicate keys were added, this may need to be called multiple times.
+  ## .. warning:: If duplicate keys were added (via now deprecated `add` proc),
```
Comment (konsumlamm) [inline lib/pure/collections/tables.nim]:
```suggestion
  ## .. warning:: If duplicate keys were added (via the now deprecated `add` proc),
```
Ditto for the other cases.

File: lib/std/enumutils.nim [hunk 3]
```diff
@@ -79,8 +79,13 @@ macro enumNames(a: typed): untyped =
...
   runnableExamples:
-    type A = enum a0 = 2, a1 = 4, a2
-    type B[T] = enum b0 = 2, b1 = 4
+    type A = enum
+      a0 = 2
+      a1 = 4
+      a2
+    type B[T] = enum
+      b0 = 2
+      b1 = 4
```
Comment (konsumlamm) [inline lib/std/enumutils.nim]:
```suggestion
    type
      A = enum
        a0 = 2
        a1 = 4
        a2
      B[T] = enum
        b0 = 2
        b1 = 4
```

File: lib/std/enumutils.nim [hunk 4]
```diff
@@ -79,8 +79,13 @@ macro enumNames(a: typed): untyped =
 iterator items*[T: HoleyEnum](E: typedesc[T]): T =
   ## Iterates over an enum with holes.
   runnableExamples:
-    type A = enum a0 = 2, a1 = 4, a2
```
Comment (timotheecour) [inline lib/std/enumutils.nim:82]: previous way in 1 line was fine

File: lib/std/enumutils.nim [hunk 5]
```diff
@@ -130,9 +135,19 @@ template symbolRank*[T: enum](a: T): int =
   ## for small enums, otherwise is `O(T.enumLen)`.
   runnableExamples:
     type
-      A = enum a0 = -3, a1 = 10, a2, a3 = (20, "f3Alt") # HoleyEnum
```
Comment (timotheecour) [inline lib/std/enumutils.nim:133]: ditto, previous way in 1 line is fine

File: lib/pure/asyncdispatch.nim (hunk 1 above)
Comment (narimiran) [reply lib/pure/asyncdispatch.nim:50]: As one wise man below said: "previous way was fine" :P

Comment (konsumlamm) [reply lib/pure/asyncdispatch.nim:50]: What? There are clear advantages of `runnableExamples` (see the contributing guidelines), it's not just different formatting.

**Pull request merged**